### ISSUE TITLE
App Store Connect upload failure with ERROR: Asset validation failed (90049) This bundle is invalid. (`_Datadog_Private`)

### ISSUE THREAD

**Issue by alexfanatics** [DataDog/dd-sdk-ios#1303]
Greetings!

I'm using DataDog version `1.20.0` with Swift PM .

When archiving our app and delivering to App Store Connect via `fastlane pilot`, our upload completes before being rejected for the following reason:

> ```ERROR: Asset validation failed (90049) This bundle is invalid. The bundle at path Payload/my-app-name.app/Frameworks/_Datadog_Private.framework has an invalid CFBundleIdentifier '_Datadog_Private' There are invalid characters(characters that are not dots, hyphen and alphanumerics) that have been replaced with their code point '\u005fDatadog\u005fPrivate' CFBundleIdentifier must be present, must contain only alphanumerics, dots, hyphens and must not end with a dot. [see the Core Foundation Keys at https://developer.apple.com/library/content/documentation/General/Reference/InfoPlistKeyReference/Articles/CoreFoundationKeys.html#//apple_ref/doc/uid/TP40009249-102070-TPXREF105]```

After inspecting the ipa, it does seem to be an invalid `CFBundleIdentifier` based on the apple documentation the ITMS error specified at that path (`Payload/my-app-name.app/Frameworks/_Datadog_Private.framework/Info.plist`):
<img width="712" alt="Screenshot 2023-06-06 at 2 30 03 PM" src="https://github.com/DataDog/dd-sdk-ios/assets/121891059/e86d8b48-f1be-428c-9a81-4247f054d50e">

It seems like the [Firebase SDK faced a similar issue about 1 year ago](https://github.com/firebase/firebase-ios-sdk/issues/9912), and [they renamed their targets to not include underscores](https://github.com/google/GoogleUtilities/pull/83).

Is anyone else seeing this when they upload their app to ASC?

**Comment by alexfanatics** [comment]
As a test, I forked this repo and renamed all occurrences of `_Datadog_Private` to `DatadogPrivate`, and my app was accepted by ASC without and validation issues. I can open a PR if the team thinks this is the right path forward. 

https://github.com/alexfanatics/dd-sdk-ios/commit/7b32c423939897d6cb16efc436e2c83431057534

**Comment by maxep** [comment]
Hey @alexfanatics! Thank you very much for the report and for trying with your fork 👍 


`_Datadog_Private` has been there for a while and it's the first report we've got, so there might be something specific with the project config that triggered this. I have a couple of questions to clarify the context:
- Is it the first time you submit your application with the Datadog SDK?
- What SPM library do you link to your project? `Datadog`, `DatadogStatic`, or `DatadogDynamic`?

That being said, it's definitely something to address on our side, and renaming the target looks reasonable.

**Comment by alexfanatics** [comment]
👋 Thanks for the quick response @maxep !

> Is it the first time you submit your application with the Datadog SDK?

No, we've had Datadog integrated for quite some time. We haven't seen this response from ITMS until last Friday. 
>  What SPM library do you link to your project? Datadog, DatadogStatic, or DatadogDynamic?

We link `Datadog` through SPM directly to the app target.

Our project's simplified `xcodegen` configuration is as follows: 
```yml
name: my-app-name
targets:
  my-app-target:
    type: application
    platform: iOS
    deploymentTarget: 15.0
    info:
      ...
    sources: 
      ...
      base:
        ENABLE_BITCODE: NO
        ENABLE_ON_DEMAND_RESOURCES: YES
        SUPPORTED_PLATFORMS: iphoneos iphonesimulator
        SUPPORTS_MACCATALYST: NO
        SUPPORTS_MAC_DESIGNED_FOR_IPHONE_IPAD: NO
        SWIFT_EMIT_LOC_STRINGS: YES
        TARGETED_DEVICE_FAMILY: 1
        VERSIONING_SYSTEM: apple-generic
        ONLY_ACTIVE_ARCH: true
    dependencies:
      - package: Datadog
```

**Comment by maxep** [comment]
Thank you, @alexfanatics, all clear 👍
We will address this issue on our side, I will keep you posted.

**Comment by maxep** [comment]
Hey @alexfanatics 👋 

The changes from your fork were merged and it was released in [1.21.0](https://github.com/DataDog/dd-sdk-ios/releases/tag/1.21.0). 
Thank you for your help, very much appreciated 🙏
